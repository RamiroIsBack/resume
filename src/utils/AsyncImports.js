import React from "react";
import Loadable from "react-loadable";

export const AsyncModalWorkContainer = Loadable({
  loader: () => import("../components/containers/ModalWorkContainer")
});
export const AsyncSummaryContainer = Loadable({
  loader: () => import("../components/containers/SummaryContainer"),
  loading() {
    return <div>Loading...</div>;
  }
});
export const AsyncJobsLayout = Loadable({
  loader: () => import("../components/layout/JobsLayout"),
  loading() {
    return <div>Loading...</div>;
  }
});
export const AsyncServicesContainer = Loadable({
  loader: () => import("../components/containers/ServicesContainer"),
  loading() {
    return <div>Loading...</div>;
  }
});
export const AsyncTimeLineContainer = Loadable({
  loader: () => import("../components/containers/TimeLineContainer"),
  loading() {
    return <div>Loading...</div>;
  }
});
export const AsyncSideBarContainer = Loadable({
  loader: () => import("../components/containers/SideBarContainer"),
  loading() {
    return <div>Loading...</div>;
  }
});
export const AsyncFooterContainer = Loadable({
  loader: () => import("../components/containers/FooterContainer"),
  loading() {
    return <div>Loading...</div>;
  }
});
export const AsyncNombreContainer = Loadable({
  loader: () => import("../components/containers/NombreContainer"),
  loading() {
    return <div>Loading...</div>;
  }
});
