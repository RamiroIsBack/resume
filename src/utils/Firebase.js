import firebase from 'firebase';

var config = {
  apiKey: 'AIzaSyDF1G-uxr_Sg4jHLZAgyuVXsuHnTt8bhzU',
  authDomain: 'resume-40a8a.firebaseapp.com',
  databaseURL: 'https://resume-40a8a.firebaseio.com',
  projectId: 'resume-40a8a',
  storageBucket: 'resume-40a8a.appspot.com',
  messagingSenderId: '714638486475'
};
firebase.initializeApp(config);
const database =firebase.database()
let copyList = [];

const getCopy = (params, actionType) => {
  return dispatch => database.ref('copy').once('value')
    .then(snapshot => {
      snapshot.forEach(function(childSnapshot){
        const valor = childSnapshot.val();
        valor.id = childSnapshot.key;
        copyList.push(valor);
      });
      if (actionType != null){
        dispatch({
          type: actionType,
          params: params, // can be null
          data: copyList, // list with all d objects
        });
      }

      return snapshot.val();
    })
    .catch(err => {
      console.log('pics in db didnt download correctly');
      throw err;
    });
};

export default {
  getCopy:getCopy,
};
