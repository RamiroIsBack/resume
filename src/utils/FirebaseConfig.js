export const firebaseConfig = {
  apiKey: "AIzaSyDF1G-uxr_Sg4jHLZAgyuVXsuHnTt8bhzU",
  authDomain: "resume-40a8a.firebaseapp.com",
  databaseURL: "https://resume-40a8a.firebaseio.com",
  projectId: "resume-40a8a",
  storageBucket: "resume-40a8a.appspot.com",
  messagingSenderId: "714638486475"
};
