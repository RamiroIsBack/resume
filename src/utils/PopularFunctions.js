export default {

  figureOutOpacity: (props) =>{
    let visual = {
      opacity: 0,
    };
    if(props.section){
      if(props.section.imgLoaded){
        visual = {
          opacity: 1,
        };
      }
    }
    return visual;
  },
  selectSpecificCopy: (props,who) =>{
    let copy = {};
    if(props.copy){
      if (props.copy.copyList.length != 0){
        for(let i = 0; i<props.copy.copyList.length; i++){
          if(props.copy.copyList[i].nombre === who){
            copy = props.copy.copyList[i];
            break;
          }

        }

      }
    }
    return copy;
  },

  getOffset:(el) => {
    el = el.getBoundingClientRect();
    return {
      left: el.left + window.scrollX,
      top: el.top + window.scrollY
    };
  },

};
