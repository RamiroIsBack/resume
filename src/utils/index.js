import Firebase from './Firebase.js';
import PopularFunctions from './PopularFunctions.js';
export {
  Firebase,
  PopularFunctions,
};