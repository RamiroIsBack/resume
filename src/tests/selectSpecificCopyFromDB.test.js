/**
 * @jest-environment node
 */

import React from 'react';
import renderer from 'react-test-renderer';

import {Firebase,PopularFunctions} from '../utils';


it('selectSpecificCopyfromDB', () => {
  //only work with promises!!
  return Firebase.getCopyForTesting()
    .then(result => { //the return is a promise: result(copyList)
      let props = {
        copy:{
          copyList: result
        }
      };
      expect (PopularFunctions.selectSpecificCopy(props,'summaryPic')).toMatchSnapshot();

    });
});
