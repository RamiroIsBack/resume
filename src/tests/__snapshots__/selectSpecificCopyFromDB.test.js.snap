// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectSpecificCopyfromDB 1`] = `
Object {
  "bio": "I've always been trying to work in what I enjoy, 
 I enjoy very much programming and that is what I've mostly done,creating mobile Apps, web apps on my own or collaborating in different projects.
I also enjoyed learning from various work fields and living in different countries what made me more versatile as a worker, increasing my skills.
Nowadays I live in Baltimore U.S.A. looking forward to continue my career as web-app developer here.
At the moment I'm finishing up an e-commerce web-app project for a client in Europe.
I'm very excited for new projects!",
  "id": "summaryPic",
  "nombre": "summaryPic",
  "nombrePic": "yoClimbPortrait.jpg",
  "urlPic": "https://firebasestorage.googleapis.com/v0/b/resume-40a8a.appspot.com/o/fotos%2FyoClimbPortrait.jpg?alt=media&token=a4ec019f-789f-4d8a-b88d-0c6f7f38507c",
}
`;
